makefiles/tools/esptool.inc.mk

```diff
@@ -10,10 +10,14 @@ BOOTLOADER_BIN = bootloader$(BOOTLOADER_COLOR)$(BOOTLOADER_INFO).bin
 ESPTOOL ?= $(RIOTTOOLS)/esptool/esptool.py
 
 # The ELFFILE is the base one used for flashing
+ELFFILE ?= $(BINDIR)/$(APPLICATION).elf
 FLASHFILE ?= $(ELFFILE)
 
-# Convert .elf to .bin before flashing
-FLASHDEPS += esp-image-convert
+# Convert .elf and .csv to .bin files at build time. These are needed for
+# flashing only, but if they are added to FLASHDEPS instead a "flash-only"
+# target would cause a rebuild of $(ELFFILE) since these files depend on it and
+# ELFFILE depends on FORCE.
+BUILD_FILES += $(FLASHFILE).bin $(BINDIR)/partitions.bin
```

Comment:
I changed these to use BUILD_FILES instead of FLASHDEPS even if they are only needed for flashing. The real problem here is that ELFFILE depends on FORCE so we can't have FLASHDEPS depend on the ELFFILE (FLASHFILE) and not trigger a rebuild. I think ELFFILE shouldn't depend on FORCE but instead have some build config dependency on a lazysponge output or similar... but that's for another bug.

Reply:
I changed this back. Turns out this doesn't work either because the CI doesn't copy those files.

I don't see a way to set this up correctly because ELFFILE is FORCEd (not sure why). I filed #16385  to address that.

Meanwhile I implemented a somewhat hacky fix setting these as .PHONY when using flash-only. This should make the CI happy.

Reply:
> Turns out this doesn't work either because the CI doesn't copy those files.

you mean when running the tests? Try adding needed files to `TEST_EXTRA_FILES`.

Reply:
@iosabi I took a look and I think adding to `BUILD_FILES` is the way to go. Simply add also a line with

```
TEST_EXTRA_FILES += $(FLASHFILE).bin $(BINDIR)/partitions.bin
```

Just below when you add to `BUILD_FILES`

Reply:
Adding them to `TEST_EXTRA_FILES` works in the CI but the following doesn't work:

```make -C tests/od BOARD=esp32-wroom-32 all flash-only -j```

however, the following works:

```make -C tests/od BOARD=esp32-wroom-32 flash -j```

which I guess is good enough. I believe that `flash -j` works because `flash: all` dependency which `flash-only` doesn't have, and `all` (well `link`) eventually depends on the `$(BUILD_FILES)`.